src/ReactiveUI.Avalonia/AutoSuspendHelper.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2024 .NET Foundation and Contributors. All rights reserved.
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+using System;
+using System.Reactive;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Reactive.Subjects;
+using System.Threading;
+using Avalonia.Controls;
+using Avalonia.Controls.ApplicationLifetimes;
+using Splat;
+
+namespace ReactiveUI.Avalonia;
+
+/// <summary>
+/// A ReactiveUI AutoSuspendHelper which initializes suspension hooks for
+/// Avalonia applications. Call its constructor in your app's composition root,
+/// before calling the RxApp.SuspensionHost.SetupDefaultSuspendResume method.
+/// </summary>
+public sealed class AutoSuspendHelper : IEnableLogger, IDisposable
+{
+    private readonly Subject<IDisposable> _shouldPersistState = new Subject<IDisposable>();
+    private readonly Subject<Unit> _isLaunchingNew = new Subject<Unit>();
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AutoSuspendHelper"/> class.
+    /// </summary>
+    /// <param name="lifetime">Pass in the Application.ApplicationLifetime property.</param>
+    public AutoSuspendHelper(IApplicationLifetime lifetime)
+    {
+        RxApp.SuspensionHost.IsResuming = Observable.Never<Unit>();
```

Comment:
With Avalonia 11.1 we can have this observable properly implemented. But likely should be done in dedicated PR, as well as it raises minimal Avalonia version (right now this PR targets 11.0).